File: core/src/fr/zlandorf/wildwest/bob/states/VisitBankAndDepositGold.java
package fr.zlandorf.wildwest.bob.states;

import com.badlogic.gdx.ai.fsm.State;
import com.badlogic.gdx.ai.msg.Telegram;
import fr.zlandorf.wildwest.Location;
import fr.zlandorf.wildwest.bob.Bob;

public class VisitBankAndDepositGold implements State<Bob> {
    @Override
    public void enter(Bob bob) {
        if (bob.getLocation() != Location.BANK) {
            bob.talk("Goin' to the bank. Yes siree");
            bob.setLocation(Location.BANK);
        }
    }

    @Override
    public void update(Bob bob) {
        bob.addGoldCarriedToWealth();
        bob.talk("Depositing gold. Total savings now: " + bob.getWealth());
        bob.getFsm().revertToPreviousState();
    }

    @Override
    public void exit(Bob bob) {
        bob.talk("Leavin' the bank");
    }

    @Override
    public boolean onMessage(Bob bob, Telegram telegram) {
        return false;
    }
}

File: core/src/fr/zlandorf/wildwest/entities/bob/states/VisitBankAndDepositGold.kt
package fr.zlandorf.wildwest.entities.bob.states

import com.badlogic.gdx.ai.fsm.State
import com.badlogic.gdx.ai.msg.Telegram
import fr.zlandorf.wildwest.domain.Location
import fr.zlandorf.wildwest.entities.bob.Bob

class VisitBankAndDepositGold : State<Bob> {
    override fun enter(bob: Bob) {
        if (bob.location != Location.BANK) {
            bob.talk("Goin' to the bank. Yes siree")
            bob.location = Location.BANK
        }
    }

    override fun update(bob: Bob) {
        bob.addGoldCarriedToWealth()
        bob.talk("Depositing gold. Total savings now: " + bob.wealth)
        bob.fsm.revertToPreviousState()
    }

    override fun exit(bob: Bob) {
        bob.talk("Leavin' the bank")
    }

    override fun onMessage(bob: Bob, telegram: Telegram): Boolean {
        return false
    }
}
